rfd/0153-resource-guidelines.md

```diff
@@ -369,6 +374,82 @@ func (s *FooService) UpdateFoo(ctx context.Context, foo *foov1.Foo) (*foov1.Foo,
 }
 ```
 
+#### Patch
+
+For handling partial updates of a resource, the patch operation exists. The
+patch operation accepts the name of the resource and an update function. It 
+fetches the resource, calls the update function with the fetched resource and
+then calls `ConditionalUpdate` with the updated resource. It then returns the
+updated resource with the revision generated by the backend set.
+
+By default, the revision of the fetched resource is used for the
+`ConditionalUpdate`, however, it is possible for the mutation function to set
+the revision. This allows a patch operation to be performed conditionally
+against a specific revision.
+
+This method is not directly exposed by the gRPC API but is used to implement
+the Update RPC when a field mask has been provided. In this case, the update
+function should apply the field mask.
+
+This replaces methods such as `UpdateAndSwap`.
+
+```go
+func (s *FooService) PatchFoo(
+	ctx context.Context,
+	name string,
+	updateFn func(foo *foov1.Foo) (foo *foov1.Foo, error),
+) (foo *foov1.Foo, error) {
```

Comment:
Could this be a method in the generic service implementation?

Reply:
I don't see why not.

Reply:
Is there somewhere I need to indicate this will be included in the generic service?

Reply:
Sure - I can raise a PR to include it there :) I'll raise that when I undraft this.